### PR TITLE
feat: add shell.trashItem() to replace shell.moveItemToTrash()

### LINKED ISSUE #11555: async shell.moveItemToTrash ?
* Electron version: 1.7.10
* Operating system: All

### Expected behavior

`shell.moveItemToTrash` API should have async version

### Actual behavior

`shell.moveItemToTrash` API only has sync version which is bad for Renderer process responsiveness

### PR DESCRIPTION
**Pull request by nornagon**
#### Description of Change
Closes #11555.

#### Checklist

- [x] PR description included and stakeholders cc'd
- [x] `npm test` passes
- [x] tests are [changed or added](https://github.com/electron/electron/blob/master/docs/development/testing.md)
- [x] relevant documentation is changed or added
- [x] PR title follows semantic [commit guidelines](https://github.com/electron/electron/blob/master/docs/development/pull-requests.md#commit-message-guidelines)
- [x] [PR release notes](https://github.com/electron/clerk/blob/master/README.md) describe the change in a way relevant to app developers, and are [capitalized, punctuated, and past tense](https://github.com/electron/clerk/blob/master/README.md#examples).
- [x] This is **NOT A BREAKING CHANGE**. Breaking changes may not be merged to master until 11-x-y is branched.

#### Release Notes

Notes: Added a new asynchronous `shell.trashItem()` API, replacing the synchronous `shell.moveItemToTrash()`.

### CODE REVIEW

**Comment by jkleinsc** [comment]
@electron/wg-api approved this at the August 24, 2020 meeting

File: docs/api/shell.md [hunk 1]
```diff
@@ -45,15 +45,27 @@ Returns `Promise<void>`
 
 Open the given external protocol URL in the desktop's default manner. (For example, mailto: URLs in the user's default mail agent).
 
-### `shell.moveItemToTrash(fullPath[, deleteOnFail])`
+### `shell.moveItemToTrash(fullPath[, deleteOnFail])` _Deprecated_
```
Comment (codebytere) [inline docs/api/shell.md:48]: We probably want to programmatically deprecate this too - 
```
deprecate.renameFunction('shell.moveItemToTrash', 'shell.trashItem')
```

File: shell/common/platform_util.cc [hunk 2]
```diff
@@ -0,0 +1,37 @@
...
+  content::GetUIThreadTaskRunner({})->PostTask(
+      FROM_HERE, base::BindOnce(std::move(callback), success, error));
+}
+
+void TrashItem(const base::FilePath& full_path,
+               base::OnceCallback<void(bool, const std::string&)> callback) {
+  // XXX: is continue_on_shutdown right?
+  base::ThreadPool::PostTask(FROM_HERE,
+                             {base::MayBlock(), base::WithBaseSyncPrimitives(),
+                              base::TaskPriority::USER_BLOCKING,
```
Comment (deepak1556) [inline shell/common/platform_util.cc:31]: why can't this be `USER_VISIBLE` instead ? given the api being async, are we really required to respond the user interaction immediately ?

Comment (nornagon) [reply shell/common/platform_util.cc:31]: I'm not sure what the distinction is here technically, but Chromium uses this for its "OpenItem" calls: https://source.chromium.org/chromium/chromium/src/+/master:chrome/browser/platform_util.cc;l=69;drc=14bd4f1ace32db87ec49e6c535f7910f71f1d89a

Comment (deepak1556) [reply shell/common/platform_util.cc:31]: You are right, I thought there would be a priority difference based on the definition [here](https://source.chromium.org/chromium/chromium/src/+/master:base/task/task_traits.h;l=63), but seems like they have same priority https://source.chromium.org/chromium/chromium/src/+/master:base/task/thread_pool/task_tracker.cc;l=480

**Comment by nornagon** [comment]
@deepak1556 @codebytere ping?

File: shell/common/platform_util_mac.mm [hunk 3]
```diff
@@ -142,6 +145,7 @@ bool MoveItemToTrash(const base::FilePath& full_path, bool delete_on_fail) {
   }
 
   if (!did_trash) {
+    *error = base::SysNSStringToUTF8([err localizedDescription]);
     LOG(WARNING) << "NSWorkspace failed to move file " << full_path.value()
                  << " to trash: "
                  << base::SysNSStringToUTF8([err localizedDescription]);
```
Comment (deepak1556) [inline shell/common/platform_util_mac.mm:151]:
```suggestion
                 << *error;
```

**Pull request merged**